src/core/python2js.c

```diff
@@ -164,22 +164,40 @@ EM_JS_REF(JsRef, _python2js_ucs4, (const char* ptr, int len), {
   return Hiwire.new_value(jsstr);
 });
 
+void
+PyUnicode_SetExtra(PyObject* unicode, JsRef extra);
+
+JsRef
+PyUnicode_GetExtra(PyObject* unicode);
```

Comment:
Please add some comments so people can find the patch that implements these functions later.